dissect/xfs/xfs.py

```diff
@@ -325,7 +325,10 @@ def link(self) -> str:
 
         if not self._link:
             if self.inode.di_format != c_xfs.xfs_dinode_fmt.XFS_DINODE_FMT_LOCAL and self.xfs.version == 5:
-                fh = self.open()
+                # We do not use open because for non-resident symlinks self.size does not include the symlink header
+                runs = self.dataruns()
+                symlink_size = c_xfs.xfs_dsymlink_hdr.size + self.size
```

Comment:
```suggestion
                symlink_size = len(c_xfs.xfs_dsymlink_hdr) + self.size
```
Achieves the same but is a bit more readable.